### PR TITLE
Landlord apartment upload

### LINKED ISSUE #28: As landlord, I want to add a new rental location so that I can create new listings as they become available/vacant.
**User story**

As landlord, I want to add a new rental location so that I can create new listings as they become available/vacant.

**Acceptance criteria**

After clicking a 'New Location' button, filling in the requisite information and saving, other users are able to view the new location and all it's submitted data.

**Definition of Done**

This is done when I can submit a new location and other users are able to see it.

### PR DESCRIPTION
**Pull request by mok232**
This PR closes #28 
- Login as a landlord
- Upload apartment by clicking on the 'Upload Apartment' tab in the navigation bar
- Validations for the following fields: state (should be a valid US state) and Zip code (5 digits)
- After apartment has been uploaded -> redirect to confirmation page
- View apartment by going to e.g. `location/7`
- Location view integrated to display both zillow and landlord listings

Note: Please don't merge until I fix the test for map/picture_url.

### CODE REVIEW

**Comment by Ayushi-0903** [comment]
Hey, I think you should use 'makemigrations' and fix the build.

**Comment by Ayushi-0903** [comment]
Ohh sorry I just read your comment.

File: location/models.py [hunk 1]
```diff
@@ -67,4 +68,4 @@ def __str__(self):
 
     @property
     def estimated_rent_price_for_display(self):
-        return f"${self.estimated_rent_price}"
+        return f"${self.rent_price}"
```
Comment (ab7289) [inline location/models.py]: should this be changed if the method is `estimated_rent_price...`? Or should we update the method signature to reflect what it is returning or add a different method to return the actual rent price?

File: location/views.py [hunk 2]
```diff
@@ -66,3 +68,52 @@ def review(request, pk):
...
+            image = form.cleaned_data["image"]
+            suite_num = form.cleaned_data["suite_num"]
+
+            # TODO use geocoding API to get latitude and longitude
+
+            # create or retrieve an existing location
+            loc = Location.objects.get_or_create(
+                city=city, state=state, address=address, zipcode=zipcode
+            )[
+                0
```
Comment (ab7289) [inline location/views.py]: this is a total nitpick, but having the `[0]` spread out between 3 lines is super weird.

File: location/views.py [hunk 3]
```diff
@@ -66,3 +68,52 @@ def review(request, pk):
...
+            ]  # using get_or_create avoids race condition
+
+            loc.save()
+
+            # create an apartment and link it to that location
+            apt = Apartment.objects.create(
+                suite_num=suite_num, image=image, rent_price=rent_price, location=loc
```
Comment (ab7289) [inline location/views.py]: should there be some validation on the `suite_num`? It's not a huge issue's down the line if we have multiple Apartments with the same suite No for a particular location.

File: location/models.py (hunk 1 above)
Comment (mok232) [reply location/models.py]: Good point! Completely forgot to change the method name. Regarding having 2 different methods, I discussed this earlier with Yonguk and it seemed like the default rent price is stored in the `rent_price` attribute. In either case, if we are getting data from landlord or from zillow, this field should contain the actual rental price that would be displayed in the location view.

File: location/views.py (hunk 2 above)
Comment (mok232) [reply location/views.py]: Yeah I know it's really weird. I tried changing the formatting but black formatting didn't like it.

File: location/views.py (hunk 3 above)
Comment (mok232) [reply location/views.py]: Good idea! Yes, we should check if there are any apartments with the same suite number in the current location. Added some validation for that!

File: location/forms.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    zipcode = us_forms.USZipCodeField(label="Zip Code")
+    rent_price = forms.DecimalField(
+        label="Rent Price ($)", max_digits=20, decimal_places=2
+    )
+    number_of_bed = forms.IntegerField(
+        label="Bedrooms", validators=[MinValueValidator(0)]
+    )
+
+    # Handling input files with Django
+    # https://simpleisbetterthancomplex.com/tutorial/2016/08/01/how-to-upload-files-with-django.html
```
Comment (ab7289) [inline location/forms.py]: great idea adding in the comment!

File: location/forms.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                Apartment.objects.get(suite_num=suite_num, location=loc)
+                raise forms.ValidationError(
+                    "An apartment with the same suite number already exists at this location! \
+                    If you are the landlord, you can edit the apartment details by going to your account page."
+                )
+            except Apartment.DoesNotExist:
+                return suite_num
+        except Location.DoesNotExist:
+            # if the apartment belongs to a location that does not exist, we can't have duplicate apartments
+            return suite_num
```
Comment (ab7289) [inline location/forms.py]: 👍

File: mainapp/templates/nav.html [hunk 6]
```diff
@@ -53,6 +53,11 @@
                     </form>-->
                 <ul class="navbar-nav ml-auto">
                     <!-- right aligned items -->
+                    <li class="nav-item active">
+                        {% if request.user.is_authenticated and user.user_type_string == "Landlord" %}
+                        <a class="nav-link" href="{% url 'apartment_upload' %}">Upload Apartment</a>
+                        {% endif %}
+                    </li>
```
Comment (ab7289) [inline mainapp/templates/nav.html]: 👍

**Pull request merged**